tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc

```diff
@@ -769,10 +769,22 @@ bool FusedBatchNormGradTransposer::IsTraining(
 Status FusedBatchNormGradTransposer::TransposeNode(
     TransposeContext* context, utils::MutableNodeView* node) {
   DCHECK(IsFusedBatchNormGrad(*node->node()));
-  if (!ShouldProcess(*context, *node) || !IsFanoutPortRankN(*node, 0, 4) ||
-      !IsTraining(*node)) {
+
+  if (!ShouldProcess(*context, *node) || !IsFanoutPortRankN(*node, 0, 4)) {
     return Status::OK();
   }
+
+  if (!IsTraining(*node)) {
```

Comment:
Would be worth adding a comment here explaining the reason for DML's special case.